hydrolib/core/dflowfm/tim/models.py

```diff
@@ -49,6 +50,26 @@ def _get_serializer(
     def _get_parser(cls) -> Callable[[Path], Dict]:
         return TimParser.parse
 
+    @validator("timeseries", pre=True, check_fields=True, allow_reuse=True)
+    def replace_fortran_scientific_notation_for_floats(cls, value, field):
+        for record in value:
+            if isinstance(record, dict):
+                record["time"] = FortranUtils.replace_fortran_scientific_notation(
+                    record["time"]
+                )
+                record["data"] = FortranUtils.replace_fortran_scientific_notation(
+                    record["data"]
+                )
+            elif isinstance(record, TimRecord):
```

Comment:
Summary of what we just discussed: the current design of all hydrolib-core model classes is that all model data files can be parsed into a try of Pydantic BaseModel subclasses. For next year an interesting idea is to investigate whether we can change to a hybrid form of almost all current Pydantic BaseModel subclasses, but switch to pandas DataFrames for the data intensive parts such as timeseries and polylines.
This will also bring us closer to realizing #553.